File: src/TurnoPOS.Service/Services/DepartmentService.cs
using TurnoPOS.Data.Models;
using TurnoPOS.Data.Repositories;
using TurnoPOS.Service.Interfaces;

namespace TurnoPOS.Service.Services;

public class DepartmentService(IGenericRepository repository) : IDepartmentService
{
    public async Task<Department> Create(Department department)
    {
        var result = repository.Insert(department);

        await repository.SaveAsync();
        return result;
    }

    public async Task<IList<Department>> GetAll()
    {
        var query = repository.Get<Department>(null, orderBy: x => x.OrderBy(d => d.Name));
        return await repository.ToListAsync(query);
    }

    public async Task<Department?> GetById(long id)
    {
        return await repository.GetById<Department>(id)
            ?? throw new KeyNotFoundException($"Department with id {id} not found."); ;
    }

    public async Task Update(Department department)
    {
        repository.Update(department);
        await repository.SaveAsync();
    }
}

File: src/turnopos.client/src/services/DepartmentService.ts
import axios from 'axios';
import type Department from '../models/Department';

const API_BASE_URL = '/department';

class DepartmentService {
    static async getAll(): Promise<Department[]> {
        const response = await axios.get<Department[]>(API_BASE_URL);
        return response.data;
    }

    static async getById(id: number): Promise<Department> {
        const response = await axios.get<Department>(`${API_BASE_URL}/${id}`);
        return response.data;
    }

    static async create(department: Omit<Department, 'id'>): Promise<Department> {
        const response = await axios.post<Department>(API_BASE_URL, department);
        return response.data;
    }

    static async update(department: Department): Promise<void> {
        await axios.put(API_BASE_URL, department);
    }
}

export default DepartmentService;